azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -248,37 +237,52 @@ def check_processes_in_agent_cgroup(self):
             """
             if not self.enabled():
-                return
+                return True
 
-            daemon = os.getppid()
-            extension_handler = os.getpid()
-            agent_commands = set()
-            agent_commands.update(shellutil.get_running_commands())
-            systemd_run_commands = set()
-            systemd_run_commands.update(self._cgroups_api.get_systemd_run_commands())
-            agent_cgroup = CGroupsApi.get_processes_in_cgroup(self._agent_cpu_cgroup_path)
-            # get the running commands again in case new commands were started while we were fetching the processes in the cgroup;
-            agent_commands.update(shellutil.get_running_commands())
-            systemd_run_commands.update(self._cgroups_api.get_systemd_run_commands())
-
-            unexpected = []
-            for process in agent_cgroup:
-                # Note that the agent uses systemd-run to start extensions; systemd-run belongs to the agent cgroup, though the extensions don't
-                if process in (daemon, extension_handler) or process in systemd_run_commands:
-                    continue
-                # check if the process is a command started by the agent or a descendant of one of those commands
-                current = process
-                while current != 0 and current not in agent_commands:
-                    current = self._get_parent(current)
-                if current == 0:
-                    unexpected.append(process)
-                    if len(unexpected) >= 5:  # collect just a small sample
-                        break
-            if unexpected:
-                raise UnexpectedProcessesInCGroupException(unexpected=self._format_processes(unexpected))
+            def log_message(message):
+                # Report only a small sample of errors
+                if message != self._check_processes_in_agent_cgroup_last_error and self._check_processes_in_agent_cgroup_error_count < 5:
+                    self._check_processes_in_agent_cgroup_error_count += 1
+                    self._check_processes_in_agent_cgroup_last_error = message
+                    logger.info(message)
+                    add_event(op=WALAEventOperation.CGroupsDebug, message=message)
+
+            try:
+                daemon = os.getppid()
+                extension_handler = os.getpid()
+                agent_commands = set()
+                agent_commands.update(shellutil.get_running_commands())
+                systemd_run_commands = set()
+                systemd_run_commands.update(self._cgroups_api.get_systemd_run_commands())
+                agent_cgroup = CGroupsApi.get_processes_in_cgroup(self._agent_cpu_cgroup_path)
+                # get the running commands again in case new commands were started while we were fetching the processes in the cgroup;
+                agent_commands.update(shellutil.get_running_commands())
```

Comment:
Why not just have it in a single place (Ln 261) instead of 2 places (Ln 256 and Ln 261)?

Reply:
it's try to address this scenario:

` # get the running commands again in case new commands were started while we were fetching the processes in the cgroup`

Reply:
Then why even keep keep it in the ln 256? Like Isint the call in 261 enough?

Reply:
I see, we need to capture both processes that started or completed. I'll update the comment; thanks for pointing this out.